test/Index/index_self.swift

```diff
@@ -0,0 +1,13 @@
+// RUN: %target-swift-ide-test -print-indexed-symbols -source-filename %s | %FileCheck %s
+
+struct Foo {
+  init() {} // CHECK: [[@LINE]]:3 | constructor/Swift | init() | [[init_USR:.*]] | Def,RelChild | rel: 1
+
+  static func bar() -> Self {
```

Comment:
`Self` in this context also isn't indexed yet, I figured that could potentially be done as a follow up. I also thing that one is a bit less important since the type it's referencing will always have another reference in the same file, where the initializer being referenced could be in another file